templates/contextual-services/quicksuggest-impression/quicksuggest-impression.1.schema.json

```diff
@@ -0,0 +1,58 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "title": "quicksuggest-impression",
+  "properties": {
+    @ACTIVITY-STREAM_EXPERIMENTS_1_JSON@,
+    "context_id": {
+      "description": "A UUID representing this user. Note that it's not client_id, nor can it be used to link to a client_id",
+      "type": "string",
+      "pattern": "^\\{[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\\}$"
```

Comment:
What's the reason for the braces rather than just using standard uuid?  I see that activity stream pings use this too

Reply:
No particular reason, this is what Firefox UUID generator produces, and I believe it's still a legit v4 UUID.

Reply:
We already have some [special handling in the pipeline for validating `impression_id` with the curly braces](https://github.com/mozilla/gcp-ingestion/blob/500cc2d88ad3bbfbb588325b70705121e86cb7ac/ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/ParsePayload.java#L320-L334). Note that we will need some follow-ups to this to support context_id as a user identifier. We'll want to [enable this as a source for sample_id](https://github.com/mozilla/gcp-ingestion/blob/500cc2d88ad3bbfbb588325b70705121e86cb7ac/ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/ParsePayload.java#L251-L260) and we'll need to get this added to deletion-request pings and get configuration added to Shredder.

Reply:
Yeah, the "deletion-request" is considered as well in this data collection proposal. We will consult legal to see if it's OK to re-use the "impression_id" here to simplify that work.